flagsmith/models.go

```diff
@@ -85,14 +84,156 @@ func (f *FlagResourceData) ToClientFS(featureStateID int64, feature int64, envir
 	}
 }
 
-// Generate a new FlagResourceData from client `FeatureState`
-func MakeFlagResourceDataFromClientFS(clientFS *flagsmithapi.FeatureState) FlagResourceData {
+// Generate a new FeatureStateResourceData from client `FeatureState`
+func MakeFeatureStateResourceDataFromClientFS(clientFS *flagsmithapi.FeatureState) FeatureStateResourceData {
 	fsValue := MakeFeatureStateValueFromClientFSV(clientFS.FeatureStateValue)
-	return FlagResourceData{
+	return FeatureStateResourceData{
 		ID:                types.Number{Value: big.NewFloat(float64(clientFS.ID))},
 		Enabled:           types.Bool{Value: clientFS.Enabled},
 		FeatureStateValue: &fsValue,
 		Feature:           types.Number{Value: big.NewFloat(float64(clientFS.Feature))},
 		Environment:       types.Number{Value: big.NewFloat(float64(clientFS.Environment))},
 	}
 }
+
+type MultivariateOptionResourceData struct {
+	Type                        types.String `tfsdk:"type"`
+	ID                          types.Number `tfsdk:"id"`
+	UUID                        types.String `tfsdk:"uuid"`
+	FeatureID                   types.Number `tfsdk:"feature_id"`
+	FeatureUUID                 types.String `tfsdk:"feature_uuid"`
+	ProjectID                   types.Number `tfsdk:"project_id"`
+	IntegerValue                types.Number `tfsdk:"integer_value"`
+	StringValue                 types.String `tfsdk:"string_value"`
+	BooleanValue                types.Bool   `tfsdk:"boolean_value"`
+	DefaultPercentageAllocation types.Number `tfsdk:"default_percentage_allocation"`
+}
+
+func NewMultivariateOptionFromClientOption(clientMvOption *flagsmithapi.FeatureMultivariateOption) MultivariateOptionResourceData {
+	mvOption := MultivariateOptionResourceData{
+		Type:                        types.String{Value: clientMvOption.Type},
+		ID:                          types.Number{Value: big.NewFloat(float64(clientMvOption.ID))},
```

Comment:
It's the opposite actually, we are converting a bunch of integers to `big.NewFloat` because that's the type of `types.Number` struct (that terraform uses to store state)

Reply:
Ah, ok 👍